src/plugins/data/public/shard_failure_modal/shard_failure_description.tsx

```diff
@@ -96,7 +96,6 @@ export function ShardFailureDescription(props: ShardFailure) {
       <EuiFlexItem grow={false}>
         <EuiDescriptionList
           type="responsiveColumn"
-          gutterSize="s"
```

Comment:
It looks like the shard failure modal doesn't display correctly with these changes:
<img width="796" alt="shard_modal" src="https://github.com/elastic/kibana/assets/25592674/887e1fa5-72f8-4221-9701-e1231f3bd4be">

Unfortunately it's tricky to test this modal since you need shard failures, but they can be simulated for testing if you want to see it:
- Navigate to Discover and create a data view with the pattern `kibana*`.
- Go to this line in `search_source`: https://github.com/elastic/kibana/blob/501e9bda2a6c17732c525d557b97f12c48dfe345/src/plugins/data/common/search/search_source/search_source.ts#L944
- Paste this before the return line:
```ts
body.query.bool.must.push({
  error_query: {
    indices: [
      {
        name: 'kibana_sample_data_logs',
        shard_ids: [0, 1],
        error_type: 'exception',
        message: 'Testing shard failures!',
      },
    ],
  },
});
```
- Reload the page and you should see a shard failure callout with a "Show details" button to open the modal:
  <img width="1027" alt="no_results" src="https://github.com/elastic/kibana/assets/25592674/5f687565-65be-4431-93c4-1485c1c7dc33">

I'm not sure if this is related to the `gutterSize` change or something else in the upgrade, but we do have some custom styles for this in `src/plugins/data/public/shard_failure_modal/_shard_failure_modal.scss` that I'm guessing are probably related.

Reply:
@davismcphee The fix for this item has just been pushed to this branch. I tested in Chrome, Firefox, Edge, and Safari at mobile and deskop widths to kick off `dispay: block` and `display: grid` layouts.